test/git_resolver_test.cr

```diff
@@ -38,6 +38,49 @@ module Shards
       assert_equal "0.2.0", library.installed_spec.not_nil!.version
     end
 
+    def test_origin_changed
+      dependency = Dependency.new("library", {"git" => git_url("library")})
+      library = GitResolver.new(dependency)
+      library.install("0.1.2")
+
+      # Change the origin in the cache repo to https://github.com/foo/bar
+      Dir.cd(library.local_path) do
+        run "git remote set-url origin https://github.com/foo/bar"
+      end
+
+      # All of these alternatives should not trigger origin as changed
+      same_origins = [
+        "https://github.com/foo/bar",
+        "https://github.com:1234/foo/bar",
+        "http://github.com/foo/bar",
+        "ssh://github.com/foo/bar",
+        "git://github.com/foo/bar",
+        "rsync://github.com/foo/bar",
+        "git@github.com:foo/bar",
+        "bob@github.com:foo/bar",
+        "github.com:foo/bar",
+      ]
+
+      same_origins.each do |origin|
+        dependency["git"] = origin
+        refute library.origin_changed?, origin
+      end
+
+      # These alternatives should all trigger origin as changed
+      changed_origins = [
+        "https://github.com/foo/bar2",
+        "https://github.com/foos/bar",
+        "https://githubz.com/foo/bar",
+        "file:///github.com/foo/bar",
```

Comment:
This array is missing a ssh+git example.

Reply:
Added tests.